Backend/Remora.Discord.Gateway/DiscordGatewayClient.cs

```diff
@@ -1047,6 +1047,16 @@ private async Task<Result> GatewayReceiverAsync(TimeSpan heartbeatInterval, Canc
 
                 var receivedAt = DateTime.UtcNow;
 
+                if (receivedPayload.Error is Results.WebSocketError { State: WebSocketState.Aborted })
+                {
+                    // request reconnect.
+                    _shouldReconnect = true;
+
+                    // idk if connection is resumable.
+                    _isSessionResumable = false;
+                    return Result.FromSuccess();
```

Comment:
I am not completely sure if it should use ``FromSuccess()``, or simply use ``FromError(receivedPayload);`` here.